### ISSUE TITLE
Fire Doors B: 120

### ISSUE THREAD

**Issue by Jibbajabbafic** [unitystation/unitystation#3027]
###  Description
They should fulfil the following criteria:
- [x] Animated opening and closing with sounds
- [x] Trigger automatically when they detect these things near them: 
  - [x] Fires or a high temperature
  - [x] Low and high pressure
- [x] Be fire resistant
- [x] Allow crowbar to open and close them manually

Other nice to have improvements:
- [x] Add a way to easily connect fire alarms to fire doors (copy the way the APCs handle connecting devices, see APCConnectingEditor.cs)
- [x] Ensure overlapping doors can still be bumped into regardless of the order fire doors were placed

Once this is done #4118 can be completed.

**Comment by The-Scout** [comment]
you should add that they should lower whenever a room gets depressurized. very important with our current atmos implimentation

**Comment by StrawberryMoses** [comment]
Depending on how hard the implementer wants to go on this- iirc on TG fire doors work with a sort of "zone" concept. The station is broken up into rooms / halls and are used for two purposes (that I can remember): electricity and fire doors. The apc powers everything in that room, and similarly, when a fire is detected or the rooms fire thingy is pulled, all of the fire doors in the room close.
Similarly, this is used when someone tries to steal the captains antique gun- when the case is broken, all fire doors in the captains office close.

**Comment by Jibbajabbafic** [comment]
Do the 'zones' have dedicated sensor objects for fires? I think just having the doors do it for now would be good enough.

**Comment by StrawberryMoses** [comment]
If I was going to guess, I'd say it would be the fire alarm control thingy, the air alarm, or the vents.
I agree, the doors doing that would probably be best / easiest for now- just a thought that might be worth keeping in mind

**Comment by Jibbajabbafic** [comment]
Ye, we can make an issue for those too

**Comment by Misterixmax** [comment]
There is air alarm on TG. It has a sensor and some other stuff, for example you can configure vents or scrubbers - it's needed to clear room from toxins, co2 or so. Also air alarm controls filters and air composition with configured % of each gas type.  And it sends data about atmos inside the room to atmos technician's room where all the station's atmos can be monitored. Also one monitor is on the bridge.

**Comment by h3killa** [comment]
Ill snag this one if no one else has

**Comment by RedlineTriad** [comment]
Neither superscout or @h3killa are responding, so I am marking this Issue as free again.

**Comment by smugmann** [comment]
can i get assigned to this so i would feel responsible?

**Comment by RedlineTriad** [comment]
@smugmann done, now get on it.

**Comment by Itsnotjames** [comment]
Working on this with @Aranclanos

**Comment by RedlineTriad** [comment]
Can't assign to Aranclanos unless he makes a comment.

**Comment by ThatDan123** [comment]
Dont worry about mapping, just get the doors working then mappers can do the tedious bit.

**Comment by Aranclanos** [comment]
I refuse to leave a comment

**Comment by RedlineTriad** [comment]
@Aranclanos and @Itsnotjames are you still working on this?

**Comment by DooblyNoobly** [comment]
@Aranclanos let me know the status of this

**Comment by Aranclanos** [comment]
It's missing air alarms sprite change and sound and theres a bug where the alarms check the pressure of the wallmount tile instead of the floor in front.
But then at the same time hacking made firelocks be unable to close on top of other doors, not taking into account that now all doors cant close due to a runtime
